src/ast/semantic_analyser.cpp

```diff
@@ -162,6 +162,22 @@ void SemanticAnalyser::visit(Builtin &builtin)
       err_ << arch::name() << " doesn't support " << builtin.ident << std::endl;
     builtin.type = SizedType(Type::integer, 8);
   }
+  else if (!builtin.ident.compare(0, 4, "sarg") && builtin.ident.size() == 5 &&
```

Comment:
This is basically identical to the "args" one right, maybe those can be merged to avoid duplicate code?

Reply:
After the most recent changes, these two else-if blocks have diverged a bit.